lib/drivers/aws/config.go

```diff
@@ -32,6 +33,32 @@ type Config struct {
 	// Optional
 	AccountIDs   []string          `json:"account_ids"`   // AWS Trusted account IDs to filter vpc, subnet, sg, images, snapshots...
 	InstanceTags map[string]string `json:"instance_tags"` // AWS Instance tags to use when this node provision them
+
+	// Manage the AWS dedicated hosts to keep them busy and deallocate when not needed
+	// Key of the map is name of the pool - will be used for identification of the pool
+	DedicatedPool map[string]DedicatedPoolRecord `json:"dedicated_pool"`
+}
+
+// Stores the configuration of AWS dedicated pool of particular type to manage
+// aws ec2 allocate-hosts --availability-zone "us-west-2c" --auto-placement "on" --host-recovery "off" --host-maintenance "off" --quantity 1 --instance-type "mac2.metal"
+type DedicatedPoolRecord struct {
+	Type string `json:"type"` // Instance type handled by the dedicated hosts pool (example: "mac2.metal")
+	Zone string `json:"zone"` // Where to allocate the dedicated host (example: "us-west-2c")
+	Max  uint   `json:"max"`  // Maximum dedicated hosts to allocate (they sometimes can handle more than 1 capacity slot)
+
+	// Optimization for the Mac dedicated hosts to send them in [scrubbing process] to save money
+	// (scrubbing is free but takes ~1-2h) when we can't release the host ([24h min limit]). When
+	// this option is set to 0 - no scrubbing is enabled. When it's set - then it's duration to
+	// stay idle and then run and terminate empty instance to trigger scrubbing. Without this delay
+	// the host will have no time to be utilized by the new workload, which is not good from
+	// utilization perspective.
+	//
+	// Current implementation is attached to state update, which could be API consuming, so this
+	// duration should be >= 1 min, otherwise API requests will be too often.
+	//
+	// [scrubbing process]: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/mac-instance-stop.html
+	// [24h min limit]: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-mac-instances.html#mac-instance-considerations
+	ScrubbingDelay util.Duration `json:"scrubbing_delay"`
```

Comment:
Just rewrote a bit in d1773628c9931ac2a0a9a3e6a7b498d16a924ee9, maybe that will be better?

Reply:
How about this:

```
// Amount of time to delay re-utilizing a dedicated macOS metal host, used to save money.
//
// The AWS macOS metal nodes have two special characteristics:
//
// 1. Due to licensing agreements, you must utilize a host for at least 24 hours
//    due to the [24hour minimum limit].
// 2. Due to technical / virtualization needs specific to macOS, there is a
//    [scrubbing process] involved to restore a machine to initial conditions.
//    The scrubbing process is free (you are not billed when the machine is scrubbing),
//    but the process can take 1-2 hours to restore the machine.
//
// When the value is 0, there is *no* scrubbing delay.  A value of greater than 0 indicates
// how long the scrubbing delay should be.
//
// **NOTE**: in the current implementation, this value is attached to state update.  This
// involves executing API calls every ScrubbingDelay time units, so users are encouraged
// to choose a value of greater than or equal to 1 minute to avoid making API requests
// too frequently (there is a delay for the API call, and you pay for the API call).
//
// [scrubbing process]: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/mac-instance-stop.html
// [24hour minimum limit]: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-mac-instances.html#mac-instance-considerations 
```

I think it is a little closer to what we are looking for.  What I do not like is `When the value is 0, there is *no* scrubbing delay` and the `**NOTE**` after it.  What I would _like_ is to be able to say

```
// By default, the value is 0, indicating there is no scrubbing delay.  A value of greater than 0 ...
```

but it is not clear to me; what is the _default_ value for this field?

Reply:
I guess also a quick sanity check; the first sentence, as far as I understand this is correct `Amount of time to delay re-utilizing a dedicated macOS metal host`, but maybe it needs to be fixed up a little.  Point is it is close, I think, but not quite perfect!